src/service/internal_tables_access.h

```diff
@@ -357,6 +358,77 @@ namespace ccf
       return service.has_value() && service->cert == expected_service_cert;
     }
 
+    static bool endorse_previous_identity(
+      ccf::kv::Tx& tx, const ccf::crypto::KeyPair_OpenSSL& service_key)
+    {
+      auto service = tx.ro<ccf::Service>(Tables::SERVICE);
+      auto active_service = service->get();
+
+      auto previous_identity_endorsement =
+        tx.rw<ccf::PreviousServiceIdentityEndorsement>(
+          ccf::Tables::PREVIOUS_SERVICE_IDENTITY_ENDORSEMENT);
+
+      ccf::CoseEndorsement endorsement{};
+      std::vector<ccf::crypto::COSEParametersFactory> pheaders{};
+      std::vector<uint8_t> key_to_endorse{};
+
+      endorsement.endorsing_key = service_key.public_key_der();
+
+      if (previous_identity_endorsement->has())
+      {
+        const auto prev_endorsement = previous_identity_endorsement->get();
+
+        endorsement.endorsed_from =
+          prev_endorsement->endorsed_till.has_value() ?
+          prev_endorsement->endorsed_till.value() :
+          prev_endorsement->endorsed_from;
```

Comment:
```suggestion
        endorsement.endorsed_from =
          prev_endorsement->endorsed_till.value_or(prev_endorsement->endorsed_from);
```

Reply:
That does not seem right though, surely it's till + 1?

Reply:
No, `till` is exclusive